state-reconstruct-storage/src/snapshot.rs

```diff
@@ -72,77 +87,10 @@ impl SnapshotDB {
         Ok(H256::from(buffer))
     }
 
-    pub fn new_read_only(db_path: PathBuf) -> Result<Self> {
-        let mut db_opts = Options::default();
-        db_opts.create_missing_column_families(true);
-        db_opts.create_if_missing(true);
-
-        let db = DB::open_cf_for_read_only(
-            &db_opts,
-            db_path,
-            vec![METADATA, STORAGE_LOGS, INDEX_TO_KEY_MAP, FACTORY_DEPS],
-            false,
-        )?;
-
-        Ok(Self(db))
-    }
-
-    pub fn get_last_repeated_key_index(&self) -> Result<u64> {
-        // Unwrapping column family handle here is safe because presence of
-        // those CFs is ensured in construction of this DB.
-        let metadata = self.cf_handle(METADATA).unwrap();
-        Ok(
-            if let Some(idx_bytes) = self.get_cf(metadata, LAST_REPEATED_KEY_INDEX)? {
-                u64::from_be_bytes([
-                    idx_bytes[0],
-                    idx_bytes[1],
-                    idx_bytes[2],
-                    idx_bytes[3],
-                    idx_bytes[4],
-                    idx_bytes[5],
-                    idx_bytes[6],
-                    idx_bytes[7],
-                ])
-            } else {
-                self.put_cf(metadata, LAST_REPEATED_KEY_INDEX, u64::to_be_bytes(0))?;
```

Comment:
I always wondered what this was for?

Reply:
As I understood it, that would just ensure that we put in an entry if we didn't have one before. Though, since we update it after every new initial write anyways I don't think we need that